server.go

```diff
@@ -298,7 +328,7 @@ func (s *Server) ListenAndServe() error {
 
 	s.httpServer = &http.Server{
 		Addr:           fmt.Sprintf(":%d", s.Port),
-		Handler:        s,
+		Handler:        s.createHandler(),
```

Comment:
Would it make sense to have `createHandler` assign a field to the `Server` struct so we could continue to use the `ServeHTTP`?

e.g.

```go
func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
  s.handler.ServeHTTP(w, r)
})
```

I think that would let us avoid having to call `createHandler().ServeHTTP` in tests, but I don't feel strongly either way.

Reply:
Yeah, I did waver on this a bit, I had `s.handler` being assigned via `NewServer` at one point to allow `ServeHTTP` to be called directly without having to `nil` check `s.handler` on each request.

But I wasn't sure we needed to expose `ServerHTTP` directly on `Server` anyway so I went with a handler being created in `ListenAndServe`.

So I also don't feel super strongly, it seemed reasonable for tests to call `createHandler()` but I definitely see your point.